src/app/shared/Models/restaurant.ts

```diff
@@ -3,14 +3,14 @@ export interface Restaurant {
   name: string;
   description: string;
   dateOfCreation?: Date;
-  reviews?: Review[];
+  reviews: Review[];
   averageRating?: number;
 }
 
 export interface Review {
-  id: string;
-  rating: number;
-  dateOfVisit: Date;
+  id?: string;
+  rating?: number;
+  dateOfVisit?: Date;
   comment: string;
-  commentedByUserId: string;
+  commentedByUserId?: string;
```

Comment:
Keep in mind that we'll need to send this as well. Currently we are not storing the logged in user anywhere, but when that is done, we'll need to sent currently logged in user's id together with the review payload